src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9019,14 +9019,10 @@ private ExpressionSyntax ParseSubExpressionCore(Precedence precedence)
                 {
                     if (isAssignmentOperator)
                     {
-                        if (opKind == SyntaxKind.SimpleAssignmentExpression && this.CurrentToken.Kind == SyntaxKind.RefKeyword)
-                        {
-                            var refToken = this.EatToken();
-                            refToken = this.AddError(refToken, ErrorCode.ERR_UnexpectedToken, refToken.Text);
-                            opToken = AddTrailingSkippedSyntax(opToken, refToken);
-                        }
-
-                        leftOperand = _syntaxFactory.AssignmentExpression(opKind, leftOperand, opToken, this.ParseSubExpression(newPrecedence));
+                        ExpressionSyntax rhs = opKind == SyntaxKind.SimpleAssignmentExpression && CurrentToken.Kind == SyntaxKind.RefKeyword
+                            ? rhs = ParsePossibleRefExpression() 
```

Comment:
Could you add a test with C# 7.1? I suspect it will give the wrong error message.